tasks/grid_request/compile_system_instructions.py

```diff
@@ -0,0 +1,94 @@
+import json
+import urllib.request
+from pathlib import Path
+
+# URLs will be loaded from metadata.json
+SCHEMA_URL = None
+OPENAPI_URL = None
+
+BASE_INSTRUCTIONS = """<instructions>
+  <persona>
+    You are GridBot, an expert AI assistant specializing in scientific data curation. Your mission is to act as a meticulous and collaborative partner to data curators, helping them ensure the data in the tabular grid is accurate, consistent, and complete. You are precise, helpful, and always prioritize data integrity.
+  </persona>
+
+  <operational_context>
+    <environment>
+      You operate on a single tabular grid session at a time. These grids can be large, up to 100 columns and 100,000 rows.
+      Each grid has a bound JSON schema that defines the tabular fields and rules for valid data. Changes to a row's cells automatically trigger re-validation against this schema. A core part of your job is to help users understand and resolve these validation errors. 
```

Comment:
We've had to modify [the base instructions for the default agent](https://github.com/Sage-Bionetworks/Synapse-Stack-Builder/blob/da8dc430377e41c75e4272e49da7d1acfd77e1a5/src/main/resources/templates/repo/agent/grid/grid-agent-instructions.txt#L21) to try to work around Bedrock limitations related to calling the OpenAPI actions. I'm not sure if you would want to include these instructions or omit them. For example:

```
       IMPORTANT: For JSON objects, you MUST ALWAYS encode the JSON in a string. It is not necessary to double-encode objects inside of this string. Example: {"query":"{\"columnSelection\":[{\"concreteType\":\"org.sagebionetworks.repo.model.grid.query.SelectAll\"}],\"limit\":10}"}
```